src/vsl/parser/grammar/expr.ne

```diff
@@ -27,33 +24,43 @@ function recursiveProperty(head, tail, optional, location) {
 # like imo we should in normal functions bc even though snek does it i think is good
 # hmm we should make it property
 # also why is it not parsing correctly halp very much
-CommandChain -> Property CommandChainPart:+ {% (d, l) => recursiveProperty(d[0], d[1], false, l) %}
+CommandChain -> Property CommandChainPart:+ {% (d, l) => recursiveProperty(d[0], d[1], l) %}
 
 # TODO: optional (foo? bar == foo?.bar)
-CommandChainPart -> (delimited[ArgumentCall, ","] {% id %}):? "{" CodeBlock[Expression] "}" {% (d, l) => new t.FunctionCall(null, (d[0][0] || []).concat([d[2]]), false, l) %}
-                  | delimited[ArgumentCall, ","] {% (d, l) => new t.FunctionCall(null, d[0][0] || [], false, l) %}
-                  | Property {% (d, l) => new t.PropertyExpression(null, d[0], false, l) %}
+CommandChainPart -> (ArgumentCallHead ("," _ ArgumentCall {% nth(2) %}):* {% d => [d[0]].concat(d[1]) %}):? "{" CodeBlock[Expression] "}" {% (d, l) => new t.FunctionCall(null, d[0].concat([d[2]]), false, l) %}
+                  | ArgumentCallHead ("," _ ArgumentCall {% nth(2) %}):+ {% (d, l) => new t.FunctionCall(null, [d[0]].concat(d[1]), false, l) %}
+                  | ArgumentCallHead {%
+                  // O: KEEP
+                    (d, l) => console.log('asadsdasfajf', t.Tuple) || d[0] instanceof t.Identifier ? new t.PropertyExpression(null, d[0], false, l) : d[0] instanceof t.Tuple ? new t.FunctionCall(null, console.log(d[0].tuple) || d[0].tuple.map(item => new t.ArgumentCall(item, null, item.position)), l) : new t.FunctionCall(null, [d[0]], false, l)
```

Comment:
These lines are waaay too long D: it seems like a lot of the grammar borks come from this issue because they're hard to read that way.